### PR TITLE
Use WindowCompat.

### PR DESCRIPTION
**Pull request by Isira-Seneviratne**
#### What is it?
- [x] Codebase improvement (dev facing)

#### Description of the changes in your PR

- Use [`WindowCompat`](https://developer.android.com/reference/androidx/core/view/WindowCompat) to replace the deprecated UI visibility flags.

#### Fixes the following issue(s)

- Fixes #

#### APK testing

The APK can be found by going to the "Checks" tab below the title. On the left pane, click on "CI", scroll down to "artifacts" and click "app" to download the zip file which contains the debug APK of this PR.

#### Due diligence
- [x] I read the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md).

### CODE REVIEW

**Review by Stypox — CHANGES_REQUESTED** [review]
In fullscreen the system back&home&recent buttons do not show up when the system ui is shown. I could reproduce on API 27 and 30. I was completely unable to test on API 22 as the player seems to crash when going fullscreen...
Search and keyboard seem to work normally.

In each place where you switched from flags to `WindowCompat`, could you explain why you expect the two code snippets to do the same thing? I.e. which flag turned into which `WindowCompat` call, ...

**Comment by Isira-Seneviratne** [comment]
> In fullscreen the system back&home&recent buttons do not show up when the system ui is shown. I could reproduce on API 27 and 30. I was completely unable to test on API 22 as the player seems to crash when going fullscreen...

I didn't encounter that issue. I'll check again from my end, thanks.

> Search and keyboard seem to work normally.
> 
> In each place where you switched from flags to `WindowCompat`, could you explain why you expect the two code snippets to do the same thing? I.e. which flag turned into which `WindowCompat` call, ...

Sure, the `setDecorFitsSystemWindows` call replaces the `SYSTEM_UI_FLAG_LAYOUT_STABLE`, `SYSTEM_UI_FLAG_LAYOUT_HIDE_NAVIGATION` and `SYSTEM_UI_FLAG_LAYOUT_FULLSCREEN` flags (the compat call uses them on API 16-29) and the insets controller calls replace the system bar immersive flags.

https://developer.android.com/develop/ui/views/layout/immersive

**Comment by Isira-Seneviratne** [comment]
My bad, I set it to show only the status bar in the player UI. I'll show the navbar as well.

**Comment by Stypox** [comment]
That's not true, see this screenshot from nightly-26:

**Comment by Isira-Seneviratne** [comment]
> That's not true, see this screenshot from nightly-26: 

Okay, thanks. I checked on an API 26 emulator and it didn't show the navigation bar. I'll update the flag.

**Review by Stypox — CHANGES_REQUESTED** [review]
Everything else seems to work fine now though

File: app/src/main/java/org/schabi/newpipe/fragments/list/search/SearchFragment.java [hunk 1]
```diff
@@ -662,15 +662,14 @@ private void showKeyboardSearch() {
         if (DEBUG) {
             Log.d(TAG, "showKeyboardSearch() called");
         }
-        KeyboardUtil.showKeyboard(activity, searchEditText);
+        KeyboardUtil.showKeyboard(getActivity(), searchEditText);
```
[the remark this answers is not in the document]
Comment (Isira-Seneviratne) [reply app/src/main/java/org/schabi/newpipe/fragments/list/search/SearchFragment.java:665]: That's weird. I checked on API 23 and 31, and the keyboard showed up on both.

[the remark this answers is not in the document]
Comment (Stypox) [reply app/src/main/java/org/schabi/newpipe/fragments/list/search/SearchFragment.java:665]: I used API 33, and the keyboard itself wouldn't open

File: app/src/main/java/org/schabi/newpipe/util/KeyboardUtil.java [hunk 2]
```diff
@@ -20,36 +21,20 @@ public static void showKeyboard(final Activity activity, final EditText editText
...
-                 * a incorrect state e.g. when animations are disabled or the unfocus event of the
-                 * previous view arrives in the wrong moment (see #7647 for details).
-                 * The invalid state can be fixed by to re-focusing the editText.
-                 */
-                editText.clearFocus();
-                editText.requestFocus();
-
-                // Try again
-                imm.showSoftInput(editText, InputMethodManager.SHOW_FORCED);
-            }
```
Comment (Stypox) [inline app/src/main/java/org/schabi/newpipe/util/KeyboardUtil.java:39]: Are you sure we can remove this safely? #7647

Comment (Isira-Seneviratne) [reply app/src/main/java/org/schabi/newpipe/util/KeyboardUtil.java]: Thanks for the linked issue. The issue reoccurred on an API 23 emulator, so I reverted it. Only the system UI visibility flags have been replaced.

**Review by Stypox — APPROVED** [review]
Thanks for the patience

**Pull request merged**